openlibrary/plugins/openlibrary/js/image_change.js

```diff
@@ -0,0 +1,77 @@
+export function initChangeImageImport() {
+
+    // Pull data from data-config of class "manageCovers" in covers/manage.html
+    var data_array = $('.manageCovers').attr('data-config').split(',');
+    var doc_type_key = data_array[0];
+    var add_url = data_array[3];
+    var manage_url = data_array[4];
+
+    function add_iframe(selector, src) {
+        $(selector)
+            .append('<iframe frameborder="0" height="450" width="580" marginheight="0" marginwidth="0" scrolling="auto"></iframe>')
+            .find('iframe')
+            .attr('src', src);
+    }
+
+    // Add iframes lazily when the popup is loaded.
+    // This avoids fetching the iframes along with main page.
+    $('.coverPop')
+        .bind('click', function () {
+            // clear the content of #imagesAdd and #imagesManage before adding new
+            $('#imagesAdd').html('');
+            $('#imagesManage').html('');
+            if (doc_type_key === '/type/work') {
+                $('#imagesAdd').prepend('<div class="throbber"><h3>$_("Searching for covers")</h3></div>');
+            }
+            setTimeout(function () {
+                // add iframe to add images
+                add_iframe('#imagesAdd', add_url);
+                // add iframe to manage images
+                add_iframe('#imagesManage', manage_url);
+            }, 0);
+        })
+        .bind('cbox_cleanup', function () {
+            $('#imagesAdd').html('');
+            $('#imagesManage').html('');
+        });
+
+}
+
+export function updateImage(image) {
+    // Pull data from data-config of class "manageCovers" in covers/manage.html
+    var data_array = $('.manageCovers').attr('data-config').split(',');
+    var doc_type_key = data_array[0];
```

Comment:
this duplicates code on L42, so let's create a helper function called `getCoverData` to return a JSON with this information.

Reply:
Wouldn't `getCoverData` only be a single return statement though? I ended up getting rid of the `updateImage` function instead if that's alright. Then, I just declared all of the variables needed from JSON at the top of `initCoversChange`.